layouts/events.hbs

```diff
@@ -0,0 +1,188 @@
+                border-radius: 3px; 
+                margin-right: 8px;
+              }
+              .key-conference::before {
+                content: ""; 
+                width: 12px; 
+                height: 12px; 
+                background: #3887be; 
+                display: inline-block; 
+                border-radius: 3px; 
+                margin-right: 8px;
+              }
+              .events #map {
+                margin: 0; 
+                padding: 0;
+              }
+              </style>
+              <div id="map" class="map"></div>
+              <div class="key">
+                <span class="key-meetup" data-i18n="events-meetu">Meetup</span>
+                <span class="key-nodeschool" data-i18n="events-nodeschool">NodeSchool</span>
+                <span class="key-conference" data-i18n="events-conference">Conference</span>
+              </div>  
+              <script>
+              function haversine() {
+                var radians = Array.prototype.map.call(arguments, function(deg) { return deg/180.0 * Math.PI; });
+                var lat1 = radians[0], lon1 = radians[1], lat2 = radians[2], lon2 = radians[3];
+                var R = 6372.8; // km
+                var dLat = lat2 - lat1;
+                var dLon = lon2 - lon1;
+                var a = Math.sin(dLat / 2) * Math.sin(dLat /2) + Math.sin(dLon / 2) * Math.sin(dLon /2) * Math.cos(lat1) * Math.cos(lat2);
+                var c = 2 * Math.asin(Math.sqrt(a));
+                return R * c;
+              }    
+                  
+              L.mapbox.accessToken = 'pk.eyJ1IjoibWlrZWFsIiwiYSI6ImNpaGY5ajk5ZTA0ZTN0cmo3MzZ5NnV5eHIifQ.YBfpOTUwcbaZrNvAfDaDcQ';
+              var map = L.mapbox.map('map', 'mapbox.streets-basic')
+              var featureLayer = L.mapbox.featureLayer()
+              
+              featureLayer.on('ready', function () {
+                if ("geolocation" in navigator) {
+                  /* geolocation is available */
+                  navigator.geolocation.getCurrentPosition(function(position) {
+                    var lat = position.coords.latitude
+                      , lon = position.coords.longitude
+                      ;
+
+                    var points = featureLayer.getGeoJSON().features.map(function (feature) {
+                      var _lat = feature.geometry.coordinates[1]
+                        , _lon = feature.geometry.coordinates[0]
+                        ;
+                      return [feature.geometry.coordinates[0], feature.geometry.coordinates[1], haversine(lat, lon, _lat, _lon)]
+                    })
+                    .sort(function (a, b) {              
+                      return a[2] > b[2]
+                    })
+                    
+                    function _bounds () {
+                      for (var i=0;i<points.length;i++) {
+                        var hav = points[i][2]
+                        if (i > 10 && hav > .1) {
```

Comment:
Here's how I do bounds detection, it could be a regular `for` loop as well, though where is the fun in that?

```js
function getBounds(points) { 
	return points.reduce(function(bounds, point) {
		bounds.minLat = bounds.minLat < point.lat ? bounds.minLat : point.lat; 
		bounds.maxLat = bounds.maxLat < point.lat ? point.lat : bounds.maxLat; 
		bounds.minLon = bounds.minLon < point.lon ? bounds.minLon : point.lon; 
		bounds.maxLon = bounds.maxLon < point.lon ? point.lon : bounds.maxLon; 
		return bounds;
	}, {minLat: 180, maxLat: -180, minLon: 90, maxLon: -90}) 
}

```